src/main/kotlin/net/atos/zac/smartdocuments/SmartDocumentsService.kt

```diff
@@ -34,25 +34,32 @@ class SmartDocumentsService @Inject constructor(
     @RestClient
     private val smartDocumentsClient: SmartDocumentsClient,
 
-    @ConfigProperty(name = "SMARTDOCUMENTS_ENABLED", defaultValue = "false")
-    private val enabled: Boolean,
+    @ConfigProperty(name = "SMARTDOCUMENTS_ENABLED")
+    private val enabled: Optional<Boolean> = Optional.of(false),
```

Comment:
Why do we want to use the 'old fashioned' java.util.Optional here? We are trying to get of these in the code base.. In Kotlin you can just make this field nullable right? Null indicates it is not filled as opposed to true or false.

Reply:
Tried that, but it only works with `Optional` and with nullable we get in runtime:
```
Failed to Inject @ConfigProperty for key SMARTDOCUMENTS_CLIENT_MP_REST_URL
```

Eclipse microprofile ConfigProperty does not seem to work with nullable fileds in Kotlin and fails if there is no configuration :(

Reply:
ah.. that's unfortunate.. Maybe add a little code comment explaining why we use Optional's here to avoid this same question in future?

Reply:
This discussion seems to be similar no? https://github.com/eclipse/microprofile-config/issues/365

Reply:
Added comment